packages/plugin-ext/src/main/browser/webview/webview-resource-cache.ts

```diff
@@ -0,0 +1,88 @@
+/********************************************************************************
+ * Copyright (C) 2019 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from 'inversify';
+import { Deferred } from '@theia/core/lib/common/promise-util';
+import { MaybePromise } from '@theia/core/lib/common/types';
+
+export interface WebviewResourceResponse {
+    eTag: string | undefined,
+    body(): MaybePromise<Uint8Array>
+}
+
+/**
+ * Browser based cache of webview resources across all instances.
+ */
+@injectable()
+export class WebviewResourceCache {
+
+    protected readonly cache = new Deferred<Cache | undefined>();
+
+    constructor() {
+        this.resolveCache();
+    }
+
+    protected async resolveCache(): Promise<void> {
+        try {
+            this.cache.resolve(await caches.open('webview:v1'));
```

Comment:
Hi,

Where should 'caches' be defined?

Reply:
Good question - I didn't know either so I opened this file in [Gitpod](https://gitpod.io/#https://github.com/eclipse-theia/theia/blob/98e8bd931ce957775220ad59b9f63393484022fc/packages/plugin-ext/src/main/browser/webview/webview-resource-cache.ts) and using  "Peek Definition" from the context menu, over that symbol, I found it's from the DOM API:

![image](https://user-images.githubusercontent.com/25749063/69582451-a4144100-0fa6-11ea-99e1-5ea4423d0a31.png)

So this is implemented in/by your web browser. Hope this help.

Reply:
It’s a browser API. It’s not available over http, webviews work only over https or localhost for testing.

Reply:
Just a thought, but maybe prefixing such variables like `window.caches` would help readers?